### PR TITLE
Require config.uuid when creating PubNub instance

### PR DESCRIPTION
**Pull request by seba-aln**
refactor: Breaking change, uuid parameter is mandatory.

uuid is required parameter while creating an instance of PubNub

BREAKING CHANGES: uuid is required parameter while creating an instance of PubNub

### CODE REVIEW

File: pubnub/pnconfiguration.py [hunk 1]
```diff
@@ -36,7 +36,7 @@ def __init__(self):
         self._heartbeat_interval = PNConfiguration.DEFAULT_HEARTBEAT_INTERVAL
 
     def validate(self):
-        assert self.uuid is None or isinstance(self.uuid, str)
+        assert self.uuid and isinstance(self.uuid, str), "UUID missing or invalid type"
```
Comment (kleewho) [inline pubnub/pnconfiguration.py:39]: Is this object immutable? Is it possible to change this `uuid` to something else after we're done validating?

File: pubnub/pnconfiguration.py [hunk 2]
```diff
@@ -36,7 +36,7 @@ def __init__(self):
...
 
         if not self.uuid:
             self.uuid = utils.uuid()
```
Comment (kleewho) [inline pubnub/pnconfiguration.py:42]: I don't think we need this anymore

Comment (seba-aln) [reply pubnub/pnconfiguration.py]: true. fixed

File: pubnub/pnconfiguration.py (hunk 1 above)
Comment (seba-aln) [reply pubnub/pnconfiguration.py]: Let's leave it as it is. this change is already a small revolution. Validation of UUID is now triggered in the moment of setting it in `PNConfiguration` and during initialization of `PubNub` object

File: pubnub/pnconfiguration.py [hunk 3]
```diff
@@ -36,10 +35,10 @@ def __init__(self):
         self._heartbeat_interval = PNConfiguration.DEFAULT_HEARTBEAT_INTERVAL
 
     def validate(self):
-        assert self.uuid is None or isinstance(self.uuid, str)
+        PNConfiguration.validate_not_empty_string(self.uuid)
 
-        if not self.uuid:
-            self.uuid = utils.uuid()
+    def validate_not_empty_string(value):
+        assert value and isinstance(value, str), "UUID missing or invalid type"
```
Comment (budgetpreneur) [inline pubnub/pnconfiguration.py:41]: When I give uuid = "    ", it is not throwing exception.
To handle empty spaces, we can refactor it by adding _strip_ 
`assert value and isinstance(value, str.strip()), "UUID missing or invalid type"`

Comment (budgetpreneur) [inline pubnub/pnconfiguration.py:41]: When I give uuid = " ", it is not throwing exception.
To handle empty spaces, we can refactor it by adding strip and check length after strip.
`assert value and isinstance(value, str) and len(value.strip()) > 0 , "UUID missing or invalid type"`

**Review by kleewho — APPROVED** [review]
It looks fine for me. For what it's worth ;)

**Review by budgetpreneur — DISMISSED** [review]
LGTM

**Review by kleewho — APPROVED** [review]

**Comment by seba-aln** [comment]
@client-engineering-bot release as v6.0.0

**Review by kleewho — APPROVED** [review]

**Comment by kleewho** [comment]
@client-engineering-bot release as v6.0.0

**Review by techwritermat — DISMISSED** [review]

**Review by kazydek — APPROVED** [review]

**Review by bartk — APPROVED** [review]

**Pull request merged**